lib/ui/painting/codec.cc

```diff
@@ -362,13 +362,23 @@ MultiFrameCodec::MultiFrameCodec(std::unique_ptr<SkCodec> codec)
     : codec_(std::move(codec)) {
   repetitionCount_ = codec_->getRepetitionCount();
   frameInfos_ = codec_->getFrameInfo();
-  frameBitmaps_.resize(frameInfos_.size());
+  frameBitmaps_.clear();
+  for (const auto& frameInfo : frameInfos_) {
+    if (frameInfo.fRequiredFrame == SkCodec::kNoFrame ||
+        frameBitmaps_.count(frameInfo.fRequiredFrame) > 0) {
+      continue;
+    }
+    frameBitmaps_[frameInfo.fRequiredFrame] = SkBitmap();
+  }
   nextFrameIndex_ = 0;
 }
 
 sk_sp<SkImage> MultiFrameCodec::GetNextFrameImage(
     fml::WeakPtr<GrContext> resourceContext) {
-  SkBitmap& bitmap = frameBitmaps_[nextFrameIndex_];
+  SkBitmap rvalue;
```

Comment:
nit: this rvalue might be a bit of a confusing name, I'd be happy if we can avoid it, maybe by doing an if / else on the frame existence in the map instead of the ternary + the if on line 382? (might require making bitmap a pointer here)

Reply:
Do you think renaming it to something else, like `new_bitmap`, would be helpful? Tried a couple things here, but from what I can tell it always ends up needing to point or refer back to a named variable.

Reply:
lgtm